### PR TITLE
Forget password feature

### PR DESCRIPTION
**Pull request by henryyang42**
Add forget password feature

Some columns are added, please try to `migrate` if you want to test.

### CODE REVIEW

File: users/forms.py [hunk 1]
```diff
@@ -128,3 +128,18 @@ def is_valid(self, user):
...
+
+class UserForgetPasswordForm(forms.Form):
+    username = forms.CharField()
+    email = forms.EmailField()
+
+    def clean_email(self):
+        # Check that if username and email match or not
+        username = self.cleaned_data['username']
+        email = self.cleaned_data['email']
+        if username and email and User.objects.filter(username=username, email=email):
```
Comment (nickelcheng) [inline users/forms.py]: Is it needed to check whether the user exists or not first?

Comment (henryyang42) [inline users/forms.py]: I validate it in form.
If the form is valid, it's guaranteed that user and email corresponds to an existed user.

Comment (nickelcheng) [inline users/forms.py]: I mean, should we tell the user that he/she give us an username which does not exist?

**Comment by nickelcheng** [comment]
I got an OperationalError when I wanted to reset password.
error

File: users/forms.py (hunk 1 above)
Comment (henryyang42) [inline users/forms.py]: I think ``Username and Email don't match`` is enough XD

Comment (nickelcheng) [inline users/forms.py]: ok, that's fine. XD

**Comment by henryyang42** [comment]
It should be fixed now, please check it.

**Comment by nickelcheng** [comment]
I received an email, but after clicking the link, I got
error

**Comment by henryyang42** [comment]
I didn't encounter that problem QAQ

**Comment by nickelcheng** [comment]
It's ok now @@
However, no matter which user I use, the URL always bring me to henry's profile

screenshot from 2015-04-02 22 09 44

Both URLs redirect to this page

screenshot from 2015-04-02 22 07 47

**Comment by henryyang42** [comment]
This is because I made a terribly stupid mistake.
You should have witnessed it!


It's fixed now

**Comment by nickelcheng** [comment]
My omission QQ

**Comment by nickelcheng** [comment]
I got 403
screenshot from 2015-04-04 12 02 21

**Comment by henryyang42** [comment]
How did you get this?

**Comment by nickelcheng** [comment]
I can not re-produce the bug..
+1

**Comment by nickelcheng** [comment]
@bruce3557 please check

File: users/models.py [hunk 2]
```diff
@@ -126,6 +126,8 @@ def __unicode__(self):
 class UserProfile(models.Model):
     user = models.OneToOneField(User)
     activation_key = models.CharField(max_length=40, blank=True)
+    # default active time is 15 minutes
+    active_time = models.DateTimeField(default=datetime.now()+timedelta(minutes=15))
```
Comment (bruce3557) [inline users/models.py]:
```active_time = models.DateTimeField(default=datetime.now() + timedelta(minutes=15))

Comment (henryyang42) [inline users/models.py]: I add space around ``+`` now. However, it's suggested not to add space around ``=``when we want to indicate ``keyword``.

https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Whitespace#Whitespace

**Comment by bruce3557** [comment]
+1, going to merge

**Pull request merged**